zynqtgui/sketchpad/sketchpad_clip.py

```diff
@@ -33,13 +33,654 @@
+                self.set_graineratorSpray(float(self.getMetadataProperty("ZYNTHBOX_GRAINERATOR_SPRAY", 1)), write=False, force=True)
+                self.set_graineratorSustain(float(self.getMetadataProperty("ZYNTHBOX_GRAINERATOR_SUSTAIN", 0.3)), write=False, force=True)
+                self.set_graineratorTilt(float(self.getMetadataProperty("ZYNTHBOX_GRAINERATOR_TILT", 0.5)), write=False, force=True)
+                self.set_length(float(self.getMetadataProperty("ZYNTHBOX_LENGTH", self.clip.initialLength)), write=False, force=True)
+                self.set_loopDelta(float(self.getMetadataProperty("ZYNTHBOX_LOOPDELTA", 0.0)), write=False, force=True)
+                self.set_loopDelta2(float(self.getMetadataProperty("ZYNTHBOX_LOOPDELTA2", 0.0)), write=False, force=True)
+                self.set_pitch(float(self.getMetadataProperty("ZYNTHBOX_PITCH", self.clip.initialPitch)), write=False, force=True)
+                self.set_playbackStyle(str(self.getMetadataProperty("ZYNTHBOX_PLAYBACK_STYLE", "Zynthbox.ClipAudioSource.PlaybackStyle.LoopingPlaybackStyle")), write=False, force=True)
+                self.set_snapLengthToBeat(str(self.getMetadataProperty("ZYNTHBOX_SNAP_LENGTH_TO_BEAT", True)).lower() == "true", write=False, force=True)
+                self.set_speedRatio(float(self.getMetadataProperty("ZYNTHBOX_SPEED_RATIO", self.clip.initialSpeedRatio)), write=False, force=True)
+                self.set_startPosition(float(self.getMetadataProperty("ZYNTHBOX_STARTPOSITION", self.clip.initialStartPosition)), write=False, force=True)
+                self.set_syncSpeedToBpm(str(self.getMetadataProperty("ZYNTHBOX_SYNC_SPEED_TO_BPM", True)).lower() == "true", write=False, force=True)
+                self.updateBpmDependantValues()
+
+    @Slot()
+    def writeMetadataWithoutSoundData(self):
+        self.write(writeSoundMetadata=False)
+
+    @Slot()
+    def writeMetadataWithSoundData(self):
+        self.write(writeSoundMetadata=True)
+
+    def write(self, writeSoundMetadata=False):
+        if not self.clip.isEmpty:
+            tags = {}
+            if writeSoundMetadata:
+                # When writing sound metadata, first set updated values to the respective properties and then write
+                self.set_audioType(self.clip.channel.trackType, write=False, force=True)
+                self.set_audioTypeSettings(self.clip.channel.getAudioTypeSettings(), write=False, force=True)
+                # TODO : Metadata Check if midi recording is correct or not
+                self.set_midiRecording(self.clip.zynqtgui.sketchpad.lastRecordingMidi, write=False, force=True)
+                # TODO : Metadata Check if fetching the pattern json is correct or not
+                self.set_patternJson(Zynthbox.PlayGridManager.instance().getSequenceModel(self.clip.zynqtgui.sketchpad.song.scenesModel.selectedSequenceName).getByPart(self.clip.channel.id, self.clip.channel.selectedPart), write=False, force=True)
```

Comment:
set_patternJson expects serialised json, but here it's being passed a PatternModel instance. To serialise that model, use `Zynthbox.PlayGridManager.instance().modelToJson(<the pattern model object>)`

Reply:
Fixed in [2b14bb5](https://github.com/zynthbox/zynthbox-qml/pull/12/commits/2b14bb5fcb2a804fba8d1d95d0323f36411065e4)